### PR TITLE
helidon-mp

### PR DESCRIPTION
**Pull request by whiplash**
helidon-mp

### CODE REVIEW

File: java/helidon-mp/app.yaml
Comment (waghanza) [inline java/helidon-mp/app.yaml:1]: no need of this file 😛

File: java/helidon-mp/src/main/java/web/helidon/IndexResource.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+
+@Path("/")
+public class IndexResource {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
```
Comment (waghanza) [inline java/helidon-mp/src/main/java/web/helidon/IndexResource.java:12]: should be text plain 

If you sport any error if other implementations, do not hesitate to PR

File: java/helidon-mp/src/main/java/web/helidon/UserResource.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+
+@Path("/user")
+public class UserResource {
+
+    @Path("/{id}")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
```
Comment (waghanza) [inline java/helidon-mp/src/main/java/web/helidon/UserResource.java:16]: should be text plain 

If you sport any error if other implementations, do not hesitate to PR

File: java/helidon-mp/src/main/java/web/helidon/UserResource.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public String getMessageHandler(@PathParam("id") String id) {
+        return id;
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
```
Comment (waghanza) [inline java/helidon-mp/src/main/java/web/helidon/UserResource.java:22]: should be text plain 

If you sport any error if other implementations, do not hesitate to PR

File: java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Microprofile server properties
+server.port=3000
+server.host=0.0.0.0
+
+# Change the following to true to enable the optional MicroProfile Metrics REST.request metrics
+metrics.rest-request.enabled=false
```
Comment (waghanza) [inline java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties:6]: I think this will mess-up results

**Review by waghanza — COMMENTED** [review]
🎉 Some minor comments

File: java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties (hunk 4 above)
Comment (whiplash) [reply java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties:6]: no idea

File: java/helidon-mp/src/main/java/web/helidon/UserResource.java (hunk 3 above)
Comment (whiplash) [reply java/helidon-mp/src/main/java/web/helidon/UserResource.java:22]: Right. Let me fix this.

File: java/helidon-mp/app.yaml
Comment (whiplash) [reply java/helidon-mp/app.yaml:1]: Why do you say?

**Comment by whiplash** [comment]
@waghanza  can you check the config.yaml file?

I run it locally just like this `java -jar target/helidon-mp.webserver.jar`

File: java/helidon-mp/app.yaml
Comment (waghanza) [reply java/helidon-mp/app.yaml:1]: we do not need to create a kube for now, therefore it is in the long term roadmap

File: java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties (hunk 4 above)
Comment (waghanza) [reply java/helidon-mp/src/main/resources/META-INF/microprofile-config.properties:6]: sorry mis-read, I though that was true

**Review by waghanza — CHANGES_REQUESTED** [review]
```java
root@7a1d8a7c175f:/opt/web# /usr/bin/java -jar target/helidon-mp.webserver.jar
Error: Could not find or load main class io.helidon.Main
Caused by: java.lang.ClassNotFoundException: io.helidon.Main
```

File: java/helidon-mp/app.yaml
Comment (waghanza) [inline java/helidon-mp/app.yaml:1]: this file is required ?

Comment (whiplash) [reply java/helidon-mp/app.yaml:1]: Yes

**Comment by waghanza** [comment]
I think I see the issue. The package is not something we can run, it not a fatjar

**Comment by whiplash** [comment]
> I think I see the issue. The package is not something we can run, it not a fatjar

it should be. let me see the logs.

**Comment by waghanza** [comment]
it works in stage only container, but not muti-stage

**Comment by whiplash** [comment]
not sure how to help with this

**Comment by waghanza** [comment]
I try to fix on next run

**Comment by waghanza** [comment]
maybe https://github.com/helidon-io/helidon/blob/4.0.0/examples/quickstarts/helidon-standalone-quickstart-mp/pom.xml contains this answer

**Comment by whiplash** [comment]
can I replicate the error locally?

**Comment by whiplash** [comment]
😢 I wasn't able to replicate this locally. Sorry.

**Comment by waghanza** [comment]
No need to be 😛 

I'll close this PR, and work on micropofile version later

**Pull request closed without merging**